lib/ssl/src/ssl.erl

```diff
@@ -2830,9 +2830,14 @@ add_filter(undefined, Filters) ->
 add_filter(Filter, Filters) ->
     [Filter | Filters].
 
-maybe_client_warn_no_verify(#{verify := verify_none, log_level := LogLevel}, client) ->
-    ssl_logger:log(warning, LogLevel, #{description => "Authenticity is not established by certificate path validation",
-                                        reason => "Option {verify, verify_peer} and cacertfile/cacerts is missing"}, #{});
-maybe_client_warn_no_verify(_,_) ->
-    %% Client certificate validation is optional in TLS 
+maybe_client_warn_no_verify(OrigOpts, #{verify := verify_none, log_level := LogLevel}, client) ->
```

Comment:
I think it feels strange to bring in the `OrigOpts` (the original option list) when there already is a map with the parsed options in `SslOpts`. I think is should be enough to use the map for testing this. If the map does not contain proper information for doing this it should be fixed instead. If we have `#{verify := verify_none}` in the options map why is that not enough for taking the right decision?

Reply:
We need the original ones to know if `{verify, verify_none}` is actually set by the user. The map will contain default values. And if the default value is used we want to have the warning, only when the user explicitly sets the verify_none value, do we want to suppress the warning. The common case is that you want to have the verification. In the cases you do not I suspect the options list will be quite short anyway so I would not worry about it.

Reply:
@KennethL  Well, we can avoid it using a temporary map element.